cnf-certification-test/accesscontrol/suite.go

```diff
@@ -54,188 +54,312 @@ var (
-	ginkgo.It(testID, ginkgo.Label(tags...), func() {
-		testhelper.SkipIfEmptyAny(ginkgo.Skip, testhelper.NewSkipObject(env.Containers, "env.Containers"))
-		if env.DaemonsetFailedToSpawn {
-			ginkgo.Skip("Debug Daemonset failed to spawn skipping TestNoSSHDaemonsAllowed")
-		}
-		testNoSSHDaemonsAllowed(&env)
-	})
+	checksGroup.Add(checksdb.NewCheck(testID, tags).
+		WithSkipCheckFn(skipIfDaemonsetFailedToSpawnFn).
+		WithSkipCheckFn(skipIfNoContainersFn).
+		WithCheckFn(func(c *checksdb.Check) error {
+			testNoSSHDaemonsAllowed(c, &env)
+			return nil
+		}))
 
 	testID, tags = identifiers.GetGinkgoTestIDAndLabels(identifiers.TestPodRequestsAndLimitsIdentifier)
-	ginkgo.It(testID, ginkgo.Label(tags...), func() {
-		testhelper.SkipIfEmptyAny(ginkgo.Skip, testhelper.NewSkipObject(env.Pods, "env.Pods"))
-		testPodRequestsAndLimits(&env)
-	})
+	checksGroup.Add(checksdb.NewCheck(testID, tags).
+		WithSkipCheckFn(skipIfNoPodsFn).
+		WithCheckFn(func(c *checksdb.Check) error {
+			testPodRequestsAndLimits(c, &env)
+			return nil
+		}))
 
-	// no 1337 UID's being used by pods
 	testID, tags = identifiers.GetGinkgoTestIDAndLabels(identifiers.Test1337UIDIdentifier)
-	ginkgo.It(testID, ginkgo.Label(tags...), func() {
-		testhelper.SkipIfEmptyAny(ginkgo.Skip, testhelper.NewSkipObject(env.Pods, "env.Pods"))
-		test1337UIDs(&env)
-	})
+	checksGroup.Add(checksdb.NewCheck(testID, tags).
+		WithSkipCheckFn(skipIfNoPodsFn).
+		WithCheckFn(func(c *checksdb.Check) error {
+			test1337UIDs(c, &env)
+			return nil
+		}))
 
 	testID, tags = identifiers.GetGinkgoTestIDAndLabels(identifiers.TestServicesDoNotUseNodeportsIdentifier)
-	ginkgo.It(testID, ginkgo.Label(tags...), func() {
-		testhelper.SkipIfEmptyAny(ginkgo.Skip, testhelper.NewSkipObject(env.Containers, "env.Containers"), testhelper.NewSkipObject(env.Pods, "env.Pods"))
-		testNodePort(&env)
-	})
+	checksGroup.Add(checksdb.NewCheck(testID, tags).
+		WithSkipCheckFn(skipIfNoContainersFn).
+		WithSkipCheckFn(skipIfNoPodsFn).
+		WithCheckFn(func(c *checksdb.Check) error {
+			testNodePort(c, &env)
+			return nil
+		}))
 
 	testID, tags = identifiers.GetGinkgoTestIDAndLabels(identifiers.TestCrdRoleIdentifier)
-	ginkgo.It(testID, ginkgo.Label(tags...), func() {
-		testhelper.SkipIfEmptyAny(ginkgo.Skip, testhelper.NewSkipObject(env.Crds, "env.Crds"), testhelper.NewSkipObject(env.Roles, "env.Roles"), testhelper.NewSkipObject(env.Namespaces, "env.Namespaces"))
-		testCrdRoles(&env)
-	})
-})
+	checksGroup.Add(checksdb.NewCheck(testID, tags).
+		WithSkipCheckFn(skipIfNoCrdsFn).
+		WithSkipCheckFn(skipIfNoRolesFn).
+		WithSkipCheckFn(skipIfNoNamespacesFn).
+		WithCheckFn(func(c *checksdb.Check) error {
+			testCrdRoles(c, &env)
+			return nil
+		}))
+}
 
 func checkForbiddenCapability(containers []*provider.Container, capability string) (compliantObjects, nonCompliantObjects []*testhelper.ReportObject) {
 	for _, cut := range containers {
```

Comment:
All the capability checks that use this function are being skipped, but they passed in the main branch. The problem is that in the main branch there's no "compliantObjects" in the final report:

![image](https://github.com/test-network-function/cnf-certification-test/assets/87083379/5441788d-904f-4a11-975b-efc812723c3f)

That's because none of our test containers have any cut.SecurityContext.Capabilities field set, but the new check.SetResult() function will set the check as skipped if both lists are empty.

I'd rewrite this function like this:
```
	for _, cut := range containers {
		compliant := true

		switch {
		case cut.SecurityContext == nil:
		case cut.SecurityContext.Capabilities == nil:
		case strings.Contains(cut.SecurityContext.Capabilities.String(), capability):
			compliant = false
		}

		if compliant {
			compliantObjects = append(compliantObjects, testhelper.NewContainerReportObject(cut.Namespace, cut.Podname, cut.Name, "No forbidden capability "+capability+" detected in container", true))
		} else {
			tnf.ClaimFilePrintf("Non compliant %s capability detected in container %s. All container caps: %s", capability, cut.String(), cut.SecurityContext.Capabilities.String())
			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewContainerReportObject(cut.Namespace, cut.Podname, cut.Name, "Non compliant capability "+capability+" in container", false).AddField(testhelper.SCCCapability, capability))
		}
	}
	return compliantObjects, nonCompliantObjects
```

Reply:
Ah yes I see how the objects would never get populated.  Thanks for the code snippet!